components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -2796,33 +2817,51 @@ private String getSearchAttribute(String filterOperation, String attributeValue,
     }
 
     /**
-     * get list of roles that matches the search criteria
-     * @param filterOperation operator value
-     * @param attributeValue search value
-     * @return list of role names
-     * @throws org.wso2.carbon.user.core
-    .UserStoreException
+     * Check whether the filter attribute support filtering with the domain embedded in the attribute value.
+     *
+     * @param attributeName Attribute to filter
+     * @return True if the given attribute support embedding domain
      */
-    private String[] getRoleNames(String filterOperation, String attributeValue) throws org.wso2.carbon.user.core
-            .UserStoreException {
+    private boolean isDomainEmbeddingSupportedAttribute(String attributeName) {
 
-        String searchAttribute = getSearchAttribute(filterOperation, attributeValue, FILTERING_DELIMITER);
-        return ((AbstractUserStoreManager) carbonUM).getRoleNames(searchAttribute, MAX_ITEM_LIMIT_UNLIMITED, true,
-                true, true);
+        return attributeName.equalsIgnoreCase(SCIMConstants.UserSchemaConstants.USER_NAME_URI) || attributeName
+                .equalsIgnoreCase(SCIMConstants.CommonSchemaConstants.ID_URI) || attributeName
```

Comment:
Better to switch the comparison to avoid any possible NPEs. Ex:  `SCIMConstants.UserSchemaConstants.USER_NAME_URI.equalsIgnoreCase(attributeName)`

Reply:
Fixed: d3c216f0d03c61daf97b811d8f0f543c2f9ae871